nomad/state/state_store.go

```diff
@@ -1040,6 +1055,24 @@ func deleteNodeCSIPlugins(txn *memdb.Txn, node *structs.Node, index uint64) erro
 	return nil
 }
 
+func deleteNodeFromPlugin(txn *memdb.Txn, plug *structs.CSIPlugin, node *structs.Node, index uint64) error {
+	plug.DeleteNode(node.ID)
+	plug.ModifyIndex = index
+
+	if plug.IsEmpty() {
```

Comment:
@langmartin should we also check that there are no volumes that are using the plugin here? It seems like if we don't, then denormalized queries for volumes will blow up?

Reply:
No, I think we should go ahead here, I'll just make sure the denormalization won't blow up, it should instead just default to unhealthy. We don't need to discard the volumes because the plugins temporarily failed.